spacy/tests/test_requirements.py

```diff
@@ -0,0 +1,59 @@
+import re
+from pathlib import Path
+
+
+def test_build_dependencies(en_vocab):
+    # Check that library requirements are pinned exactly the same across different setup files.
+    libs_ignore_requirements = ["pytest", "pytest-timeout", "mock", "flake8", "jsonschema"]
+    libs_ignore_setup = ["fugashi", "natto-py", "pythainlp"]
+
+    # check requirements.txt
+    root_dir = Path(__file__).parent.parent.parent
+    req_file = root_dir / "requirements.txt"
+    req_dict = {}
+    with req_file.open() as f:
+        lines = f.readlines()
+    for line in lines:
+        line = line.strip()
+        if not line.startswith("#"):
+            lib, v = _parse_req(line)
+            if lib and lib not in libs_ignore_requirements:
+                req_dict[lib] = v
+
+    # check setup.cfg and compare to requirements.txt
+    # also fails when there are missing or additional libs
+    setup_file = root_dir / "setup.cfg"
+    with setup_file.open() as f:
+        lines = f.readlines()
+    setup_keys = set()
+    for line in lines:
+        line = line.strip()
+        if not line.startswith("#"):
+            lib, v = _parse_req(line)
+            if lib and not lib.startswith("cupy") and lib not in libs_ignore_setup:
+                req_v = req_dict.get(lib, None)
+                assert req_v is not None    # if fail: setup.cfg contains a lib not in requirements.txt
```

Comment:
I guess we might as well add a message here (and the other asserts) so the output is easier to read and understand if it fails?

```suggestion
                assert req_v is not None, "in setup.cfg but not in requirements.txt: {}".format(req_v)
```